crud/readview.lua

```diff
@@ -0,0 +1,613 @@
+        -- This option is temporary and is related to [1], [2].
+        -- [1] https://github.com/tarantool/crud/issues/236
+       -- [2] https://github.com/tarantool/crud/issues/361
+        local storages_info = fiber.self().storage.storages_info_on_select
+        iter = utils.fetch_latest_metadata_for_select(space_name, vshard_router, opts,
+                                                     storages_info, iter)
+    end
+
+    -- filter space format by plan.field_names (user defined fields + primary key + scan key)
+    -- to pass it user as metadata
+    local filtered_space_format, err = utils.get_fields_format(iter.space:format(), iter.plan.field_names)
+    if err ~= nil then
+        return nil, err
+    end
+
+
+    return {
+        metadata = table.copy(filtered_space_format),
+        rows = tuples,
+    }
+end
+
+local Readview_obj = {}
+Readview_obj.__index = Readview_obj
+
+function Readview_obj:select(space_name, user_conditions, opts)
+    opts = opts or {}
+
+    local vshard_router, err = utils.get_vshard_router_instance(opts.vshard_router)
+    if err ~= nil then
+        return nil, ReadviewError:new(err)
+    end
+
+
+    return ReadviewError:pcall(sharding.wrap_select_method, vshard_router, readview_module_call_xc,
+                             space_name, self.uuid, user_conditions, opts)
+end
+
+
+function Readview_obj:pairs(space_name, user_conditions, opts)
+    checks('table', 'string', '?table', {
+        after = '?table|cdata',
+        first = '?number',
+        batch_size = '?number',
+        use_tomap = '?boolean',
+        bucket_id = '?number|cdata',
+        force_map_call = '?boolean',
+        fields = '?table',
+        fetch_latest_metadata = '?boolean',
+
+        mode = '?vshard_call_mode',
+        prefer_replica = '?boolean',
+        balance = '?boolean',
+        timeout = '?number',
+
+        vshard_router = '?string|table',
+
+        yield_every = '?number',
+    })
+
+    opts = opts or {}
+
+    if opts.first ~= nil and opts.first < 0 then
+        error(string.format("Negative first isn't allowed for pairs"))
+    end
+
+    local vshard_router, err = utils.get_vshard_router_instance(opts.vshard_router)
+    if err ~= nil then
+        error(err)
+    end
+
+    local iterator_opts = {
+        after = opts.after,
+        first = opts.first,
+        batch_size = opts.batch_size,
+        bucket_id = opts.bucket_id,
+        force_map_call = opts.force_map_call,
+        field_names = opts.fields,
+        yield_every = opts.yield_every,
+        call_opts = {
+            mode = opts.mode,
+            prefer_replica = opts.prefer_replica,
+            balance = opts.balance,
+            timeout = opts.timeout,
+            fetch_latest_metadata = opts.fetch_latest_metadata,
+        },
+    }
+
+    local iter, err = schema.wrap_func_reload(
+            vshard_router, build_select_iterator, space_name, self.uuid, user_conditions, iterator_opts
```

Comment:
This is the only line different from the `select_module.pairs`. We definitely shouldn't make such copypaste and reuse existing code while changing it, if needed.

Reply:
The approach should be used to not copypaste select code too, not only the pairs one.

Reply:
fixed